easyfsl/samplers/task_sampler.py

```diff
@@ -80,6 +80,33 @@ def episodic_collate_fn(
                 - their labels,
                 - the dataset class ids of the class sampled in the episode
         """
+        #the input_data should be a list
+        if not isinstance(input_data, list):
+            raise TypeError(
+                "Illegal type of input."
+                "check out the type of the output of the .getitem() method of your dataset and make sure it's"
+                "a List[Tuple[Tensor, int]] or List[Tuple[Tensor, 0-dim Tensor]]."
+            )
+        #the instances inside the input_data list should be all Tuples
+        if not all(isinstance(n, tuple) for n in input_data):
+            raise TypeError(
+                "Illegal type of input."
+                "check out the type of the output of the .getitem() method of your dataset and make sure it's" 
+                "a List[Tuple[Tensor, int]] or List[Tuple[Tensor, 0-dim Tensor]]."
+            )
+        #inside the tuple there should be  List[Tuple[Tensor, int]] or List[Tuple[Tensor, 0-dim Tensor]]
+        if not all( isinstance(n[0],torch.Tensor) and isinstance(n[1],int) for n in input_data):
```

Comment:
(!) Please use explicit names for your variables to make the code easily understandable. Here I'd suggest something like 

```python
for image, label in input_data:
	# 1. Check that image is a tensor. If not, raise a specific TypeError to say that the image must be a tensor and not type(image)
	# 2. Check that label is an int. If not, check that it is a tensor. If it's neither, raise TypeError.
	# 3. Check that label is a 0-dim tensor. If not, raise a ValueError. If so, cast into int.
```

Reply:
Hey @ebennequin , thank you for your notes, I'm still learning so it's really helpful to hear feedback for my code, I think I'll have enough time to change my PR tomorrow or something, I hope that's ok with you.